File: src/main/java/no/jstien/bikeapi/tsdb/read/DevNullStationTSDBReader.java
package no.jstien.bikeapi.tsdb.read;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class DevNullStationTSDBReader implements StationTSDBReader {
    @Override
    public Map<Integer,StationHistory> queryStations(RequestFactory requestFactory) {
        Map<Integer,StationHistory> result = new HashMap<>();

        Arrays.stream(requestFactory.getStationIds()).forEach(id -> {
            result.put(id, new StationHistory(id));
        });

        return result;
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/DevNullStationTSDBReader.kt
package no.jstien.bikeapi.tsdb.read

import java.util.Arrays
import java.util.HashMap

class DevNullStationTSDBReader : StationTSDBReader {
    override fun queryStations(requestFactory: RequestFactory): Map<Int, StationHistory> {
        val result = HashMap<Int, StationHistory>()

        Arrays.stream(requestFactory.stationIds).forEach { id -> result.put(id, StationHistory(id)) }

        return result
    }
}
